### PR TITLE
ngcc: preserve whitespace configuration

### PR DESCRIPTION
**Pull request by petebacondarwin**
Tracking at FW-2013

### CODE REVIEW

**Review by gkalpak — APPROVED** [review]
LGTM 👍

File: packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts [hunk 1]
```diff
@@ -55,6 +56,7 @@ export class DecorationAnalyzer {
   private rootDirs = this.bundle.rootDirs;
   private packagePath = this.bundle.entryPoint.package;
   private isCore = this.bundle.isCore;
+  private compilerOptions = this.tsConfig !== null? this.tsConfig.options: {};
```
Comment (JoostK) [inline packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts]: 0 points for the code formatter here!

File: packages/compiler-cli/ngcc/src/main.ts [hunk 2]
```diff
@@ -143,6 +144,17 @@ export interface SyncNgccOptions {
    * Default: `false` (i.e. the manifest will be used if available)
    */
   invalidateEntryPointManifest?: boolean;
+
+  /**
+   * An absolte path to a TS config file (e.g. `tsconfig.json`) or a directory containing one that
```
Comment (devversion) [inline packages/compiler-cli/ngcc/src/main.ts]:
```suggestion
   * An absolute path to a TS config file (e.g. `tsconfig.json`) or a directory containing one that
```

**Review by JoostK — APPROVED** [review]

**Review by devversion — APPROVED** [review]
LGTM for dev-infra

File: packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts (hunk 1 above)
Comment (gkalpak) [reply packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts]: ??? 😕

Comment (JoostK) [reply packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts]: @gkalpak I've not come across a ternary that is formatted as badly as this one. The question mark as well as the colon only has whitespace to their right, it looks so weird.

I'm fine with keeping as is, for sure!

Comment (gkalpak) [reply packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts]: Wow, I didn't notice the missing whitespace and it looked fine to me 🤦‍♂

Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/analysis/decoration_analyzer.ts]: Nothing to be done in this PR about this.

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/ngcc/src/main.ts [hunk 3]
```diff
@@ -143,6 +147,18 @@ export interface SyncNgccOptions {
    * Default: `false` (i.e. the manifest will be used if available)
    */
   invalidateEntryPointManifest?: boolean;
+
+  /**
+   * An absolute path to a TS config file (e.g. `tsconfig.json`) or a directory containing one, that
+   * will be used to configure module resolution with things like path mappings, if not specified
+   * explicitly via the `pathMappings` property to `mainNgcc`.
+   *
+   * If `undefined`, ngcc will attempt to load a `tsconfig.json` file from the directory above the
```
Comment (alxhub) [inline packages/compiler-cli/ngcc/src/main.ts]: I don't think ngcc should select a tsconfig implicitly - it should either be passed explicitly, or not use one.

In Angular it's rare that the top-level tsconfig is intended for compilation.

Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/main.ts]: It turns out that in CLI apps the top level tsconfig does actually commonly contain the information we require. Also I am talking to the CLI team to see if they can apply the correct tsconfig file during the async integration.

Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/main.ts]: The CLI can pass the correct configuration: https://github.com/angular/angular-cli/pull/17295

**Pull request closed without merging**